src/osp/Active/SysRender.cpp

```diff
@@ -0,0 +1,275 @@
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+#include "SysRender.h"
+
+#include "osp/Shaders/Phong.h"
+#include "adera/Shaders/PlumeShader.h"
+#include <Magnum/Shaders/MeshVisualizer.h>
+#include "osp/Shaders/FullscreenTriShader.h"
+
+#include <Magnum/GL/Buffer.h>
+#include <Corrade/Containers/ArrayViewStl.h>
+#include <Magnum/Mesh.h>
+#include <Magnum/GL/Renderer.h>
+#include <Magnum/GL/DefaultFramebuffer.h>
+#include <Magnum/GL/Framebuffer.h>
+#include <Magnum/GL/TextureFormat.h>
+#include <Magnum/GL/Renderbuffer.h>
+#include <Magnum/GL/RenderbufferFormat.h>
+
+using osp::active::SysRender;
+using osp::active::RenderOrder_t;
+using osp::active::RenderPipeline;
+
+void SysRender::add_functions(ActiveScene& rScene)
+{
+    rScene.debug_update_add(rScene.get_update_order(), "", "", "",
+        SysRender::update);
+
+    rScene.debug_render_add(rScene.get_render_order(), "", "", "",
+        SysRender::render);
+}
+
+void SysRender::setup(ActiveScene& rScene)
+{
+    Package& resources = rScene.get_context_resources();
+
+    using namespace osp::shader;
+    using namespace adera::shader;
+    using namespace Magnum;
+
+    /* Load shaders (eventually should happen elsewhere) */
+    resources.add<Phong>("phong_shader",
+        Phong{Magnum::Shaders::Phong::Flag::DiffuseTexture});
+
+    resources.add<PlumeShader>("plume_shader");
+
+    using Magnum::Shaders::MeshVisualizer3D;
+    resources.add<MeshVisualizer3D>("mesh_vis_shader",
+        MeshVisualizer3D{MeshVisualizer3D::Flag::Wireframe
+        | MeshVisualizer3D::Flag::NormalDirection});
+
+    resources.add<FullscreenTriShader>("fullscreen_tri_shader");
+
+    /* Generate fullscreen tri for texture rendering */
+    if (resources.get<GL::Mesh>("fullscreen_tri").empty())
+    {
+        Vector2 screenSize = Vector2{GL::defaultFramebuffer.viewport().size()};
+
+        float aspectRatio = screenSize.x() / screenSize.y();
+
+        std::array<float, 12> surfData
+        {
+            // Vert position    // UV coordinate
+            -1.0f,  1.0f,       0.0f,  1.0f,
+            -1.0f, -3.0f,       0.0f, -1.0f,
+             3.0f,  1.0f,       2.0f,  1.0f
+        };
+
+        GL::Buffer surface(std::move(surfData), GL::BufferUsage::StaticDraw);
```

Comment:
std::array can't be moved from. It's equivelent to a raw c-style array on the stack.

As a result, you should probably make surfData a static constexpr variable, to give the compiler a better chance at not fucking up.

You could go a step further and make it an std::initializer_list<float> as well.